### PR TITLE
perf (RingTheory.Congruence): direct inheritance for algebraic RingCon classes

### PR DESCRIPTION
**Pull request by mattrobball**
We replace the calls to `Function.Surjective.x` for constructing instances with a more direct inheritance pattern.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

**Comment by mattrobball** [comment]
!bench

**Comment by mattrobball** [comment]
!bench

**Comment by mattrobball** [comment]
!bench

File: Mathlib/GroupTheory/Congruence.lean [hunk 1]
```diff
@@ -1305,19 +1285,26 @@ instance zpowinst : Pow c.Quotient ℤ :=
   ⟨fun x z => Quotient.map' (fun x => x ^ z) (fun _ _ h => c.zpow z h) x⟩
 #align con.has_zpow Con.zpowinst
 
+@[to_additive "A quotient of a `SubNegMonoid` by an additive congruence relation is
+a `SubNegMonoid`."]
+instance divInvMonoid : DivInvMonoid c.Quotient :=
+  { Con.monoid _, Con.hasInv _, Con.hasDiv _ with
+    zpow := fun z q => q ^ z
```
Comment (ChrisHughes24) [inline Mathlib/GroupTheory/Congruence.lean:1292]:
```suggestion
  { zpow := fun z q => q ^ z
```

**Review by ChrisHughes24 — COMMENTED** [review]
There are unnecessary eta expansions here, but that shouldn't matter after lean4#2478

bors d+

**Comment by ChrisHughes24** [comment]
!bench

**Pull request closed without merging**